### ISSUE TITLE
Bug: [no-floating-promises] allowForSafeCalls doesn't work with string TypeOrValueSpecifiers

### ISSUE THREAD

**Issue by JHawk0224** [typescript-eslint/typescript-eslint#11423]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=MYewdgzgLgBATgQzAExAWwIIQJ5mAMQFc8oBLcGAXhgRzxgAoBKKgPhgG8AoGeAUyiE4YGAAU46UhD4A6OHwggANgDc%2BDKHEJ8mXAL5cuiFOiy4CxYGXDMuQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTowAbXDYciaBOiQANMpVYV2SP3h8A7gDEO0ANKdTTAMr9uiAMLH4qDEoMGY-JgATKQBBZEImMnNYSKEOJkhdAwBdJIgAXyTUjJB0oA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const randomAsyncFunction = async () => {
  return Promise.resolve(true)
}

randomAsyncFunction()
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-floating-promises": [
      "error",
      {
        "allowForKnownSafeCalls": [
          "randomAsyncFunction"
        ]
      }
    ]
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

According to the [docs](https://typescript-eslint.io/rules/no-floating-promises/), there should be no lint issue with calling this function since it's listed in the allow list.

### Actual Result

The lint rules still say it must be dealt with.

Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator. 5:1 - 5:22
> Add void operator to ignore.
> Add await operator.

### Additional Info

I believe this has to do with `isKnownSafePromiseReturn`, it never calls `valueMatchesSomeSpecifier`.
